synapse/tests/test_cortex.py

```diff
@@ -5177,3 +5178,83 @@ async def test_tag_model(self):
             self.none(await core.callStorm('return($lib.model.tags.get(cno.cve))'))
 
             await core.nodes('[ inet:ipv4=1.2.3.4 +#cno.cve.2021.haha ]')
+
+    async def test_cortex_iterrows(self):
+
+        async with self.getTestCoreAndProxy() as (core, prox):
+            await core.addTagProp('score', ('int', {}), {})
+
+            async with await core.snap() as snap:
+
+                props = {'asn': 10, '.seen': ('2016', '2017')}
+                node = await snap.addNode('inet:ipv4', 1, props=props)
+                buid1 = node.buid
+                await node.addTag('foo', ('2020', '2021'))
+                await node.setTagProp('foo', 'score', 42)
+
+                props = {'asn': 20, '.seen': ('2015', '2016')}
+                node = await snap.addNode('inet:ipv4', 2, props=props)
+                buid2 = node.buid
+                await node.addTag('foo', ("2019", "2020"))
+                await node.setTagProp('foo', 'score', 41)
+
+                props = {'asn': 30, '.seen': ('2015', '2016')}
+                node = await snap.addNode('inet:ipv4', 3, props=props)
+                buid3 = node.buid
+                await node.addTag('foo', ("2018", "2020"))
+                await node.setTagProp('foo', 'score', 99)
+
+                node = await snap.addNode('test:str', 'yolo')
+
+                node = await snap.addNode('test:str', 'z' * 500)
+
+            badiden = 'xxx'
+            await self.agenraises(s_exc.NoSuchLayer, prox.iterPropRows(badiden, 'inet:ipv4', 'asn'))
+
+            # rows are (buid, valu) tuples
+            layriden = core.view.layers[0].iden
+            rows = await alist(prox.iterPropRows(layriden, 'inet:ipv4', 'asn'))
+
+            self.eq((10, 20, 30), tuple(sorted([row[1] for row in rows])))
+
+            await self.agenraises(s_exc.NoSuchLayer, prox.iterUnivRows(badiden, '.seen'))
+
+            # rows are (buid, valu) tuples
+            rows = await alist(prox.iterUnivRows(layriden, '.seen'))
+
+            tm = lambda x, y: (s_time.parse(x), s_time.parse(y))  # NOQA
+            ivals = (tm('2015', '2016'), tm('2015', '2016'), tm('2016', '2017'))
+            self.eq(ivals, tuple(sorted([row[1] for row in rows])))
+
+            # iterFormRows
+            await self.agenraises(s_exc.NoSuchLayer, prox.iterFormRows(badiden, 'inet:ipv4'))
+
+            rows = await alist(prox.iterFormRows(layriden, 'inet:ipv4'))
+            self.eq([(buid1, 1), (buid2, 2), (buid3, 3)], rows)
+
+            # iterTagRows
+            expect = sorted(
```

Comment:
seems to be missing the successful iterTagRows test that matches these values

Reply:
Ok added a couple of tests that use it.